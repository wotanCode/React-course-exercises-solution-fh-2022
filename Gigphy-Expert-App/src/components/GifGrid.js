//import { useState, useEffect } from "react";
// import PropTypes  from 'prop-types'

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {

  // const [images, setImages] = useState([])

  const { data:images, loading }  = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {
        loading && 'Loading'
      }

      <div className="card-grid">

        {images.map(img => (
          <GifGridItem
            key={img.id}
            {...img}
          />
        ))

        }
      </div>
    </>
  )
}

// GifGrid.PropTypes = {
//   category: PropTypes.string.isRequired,
// }