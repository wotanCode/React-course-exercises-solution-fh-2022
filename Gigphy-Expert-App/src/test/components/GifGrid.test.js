import { GifGrid } from "../../components/GifGrid";
import '@testing-library/jest-dom'
import {shallow} from 'enzyme';
import {useFetchGifs} from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
  const category = 'One Punch';

  test('Debe de mostrarse correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imÃ¡genes', () => {
    const gifs = [{
        id: 'ABC',
        url: 'https://localhost/cualquiercosa.jpg',
        title: 'Cualquier cosa'
    },
    {
        id: 'ABCD',
        url: 'https://localhost/cualquiercosa2.jpg',
        title: 'Cualquier cosa 2'
      }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  });

});