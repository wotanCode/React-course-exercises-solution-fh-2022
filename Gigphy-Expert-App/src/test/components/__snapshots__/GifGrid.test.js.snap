// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="Cualquier cosa 2"
      url="https://localhost/cualquiercosa2.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  Loading
  <div
    className="card-grid"
  />
</Fragment>
`;
